lib/utils/getFileIgnorer.js

```diff
@@ -10,25 +10,38 @@ const isPathNotFoundError = require('./isPathNotFoundError');
 const DEFAULT_IGNORE_FILENAME = '.stylelintignore';
 
 /**
- * @param {{ cwd: string, ignorePath?: string, ignorePattern?: string[] }} options
+ * @param {{ cwd: string, ignorePath?: string | string[], ignorePattern?: string[] }} options
  * @return {import('ignore').Ignore}
  */
 module.exports = function getFileIgnorer(options) {
-	const ignoreFilePath = options.ignorePath || DEFAULT_IGNORE_FILENAME;
-	const absoluteIgnoreFilePath = path.isAbsolute(ignoreFilePath)
-		? ignoreFilePath
-		: path.resolve(options.cwd, ignoreFilePath);
-	let ignoreText = '';
-
-	try {
-		ignoreText = fs.readFileSync(absoluteIgnoreFilePath, 'utf8');
-	} catch (readError) {
-		if (!isPathNotFoundError(readError)) {
-			throw readError;
+	const ignorer = ignore();
+	let ignorePath = [];
+
+	if (Array.isArray(options.ignorePath) && options.ignorePath.length > 0) {
+		ignorePath = options.ignorePath;
+	} else if (typeof options.ignorePath === 'string') {
+		ignorePath.push(options.ignorePath);
+	} else {
+		ignorePath.push(DEFAULT_IGNORE_FILENAME);
+	}
+
+	for (const ignoreFilePath of ignorePath) {
```

Comment:
[suggest] How about refactoring the code be using [`Array.prototype.flat()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) like this?

```suggestion
	const ignorePaths = [options.ignorePath || []].flat();

	if (ignorePaths.length === 0) {
		ignorePaths.push(DEFAULT_IGNORE_FILENAME);
	}

	for (const ignoreFilePath of ignorePaths) {
```

Reply:
Thanks. I updated the code you suggested because it was smarter. https://github.com/stylelint/stylelint/pull/6345/commits/d1e89820a6f3aff0d35fb61f421e3a0766faaff6